test/miner_blockchain_SUITE.erl

```diff
@@ -170,6 +170,58 @@ init_per_testcase(TestCase, Config0) ->
 end_per_testcase(_TestCase, Config) ->
     miner_ct_utils:end_per_testcase(_TestCase, Config).
 
+autoskip_on_timeout_test(Config) ->
+    %% The idea is to reproduce the following chain stall scenario:
+    %% 1. 1/2 of consensus members produce timed blocks;
+    %% 2. 1/2 of consensus members do not;
+    %% 3.
+    %%      If time-based autoskip doesn't work:
+    %%          chain halts - no more blocks are created on any of the nodes
+    %%      otherwise
+    %%          chain advances, so time-based autoskip must've worked
+
+    MinersAll = ?config(miners, Config),
+    MinersCG = miner_ct_utils:in_consensus_miners(MinersAll),
+    N = length(MinersCG),
+    ?assert(N > 1, "We have at least 2 miner nodes."),
+    M = N div 2,
+    ?assertEqual(N, 2 * M, "Even split of consensus nodes."),
+    MinersCGBroken = lists:sublist(MinersCG, M),
+
+    ct:pal("N: ~p", [N]),
+    ct:pal("MinersAll: ~p", [MinersAll]),
+    ct:pal("MinersCG: ~p", [MinersCG]),
+    ct:pal("MinersCGBroken: ~p", [MinersCGBroken]),
+
+    %% Default is 120 sesonds, but for testing we want to trigger skip faster:
+    _ = [
+        ok = ct_rpc:call(Node, application, set_env, [miner, late_block_timeout_seconds, 10], 300)
+    ||
+        Node <- MinersAll
+    ],
+
+    _ = [
+        ok = ct_rpc:call(Node, miner, hbbft_skip, [], 300)
+    ||
+        Node <- MinersCGBroken
+    ],
+    _ = [
+        ok = ct_rpc:call(Node, miner, hbbft_skip, [], 300)
+    ||
+        Node <- MinersCGBroken
+    ],
+    Node1 = hd(MinersCGBroken),
+    ok = ct_rpc:call(Node1, miner, hbbft_skip, [], 300),
+    ok = ct_rpc:call(Node1, miner, hbbft_skip, [], 300),
+    ok = ct_rpc:call(Node1, miner, hbbft_skip, [], 300),
+    ok = ct_rpc:call(Node1, miner, hbbft_skip, [], 300),
+    ok = ct_rpc:call(Node1, miner, hbbft_skip, [], 300),
+    ok = ct_rpc:call(Node1, miner, hbbft_skip, [], 300),
+
+    ok = miner_ct_utils:assert_chain_halted(MinersAll),
+    ok = miner_ct_utils:assert_chain_advanced(MinersAll, 2500, 5),
```

Comment:
I'm not following what's going on here. `assert_chain_halted` checks every 5 seconds for over a minute that height is stuck on all nodes and after all that - `assert_chain_advanced` also passes? Is it due to getting the timing right with just the right number of `hbbft_skip` calls?

Reply:
Also, why then focus on one of the nodes?

The test could use a comment explaining the idea behind this magic number of `hbbft_skip` calls.

Reply:
added some comments.   the double check is just to make sure that the chain halts for a while and then goes forward again.  I guess there might be a quicker way to figure that out?